buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/publish/GeneratePom.kt

```diff
@@ -0,0 +1,70 @@
+package org.gradle.plugins.publish
+
+import accessors.base
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ExternalDependency
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.artifacts.maven.Conf2ScopeMappingContainer
+import org.gradle.api.plugins.MavenRepositoryHandlerConvention
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.Upload
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.getting
+import org.gradle.kotlin.dsl.withConvention
+import org.gradle.kotlin.dsl.withType
+import org.gradle.kotlin.dsl.*
+import java.io.File
+
+open class GeneratePom : DefaultTask() {
+    @InputFile
```

Comment:
This is actually the output of the task, but based on the weirdness of what we're doing (we use publishCompile to configure publishRuntime), I would just drop the incremental annotations for now.